File: .pub-cache/hosted/pub.dartlang.org/camera_android-0.9.8+3/android/src/test/java/io/flutter/plugins/camera/features/exposurelock/ExposureModeTest.java
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package io.flutter.plugins.camera.features.exposurelock;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class ExposureModeTest {

  @Test
  public void getValueForString_returnsCorrectValues() {
    assertEquals(
        "Returns ExposureMode.auto for 'auto'",
        ExposureMode.getValueForString("auto"),
        ExposureMode.auto);
    assertEquals(
        "Returns ExposureMode.locked for 'locked'",
        ExposureMode.getValueForString("locked"),
        ExposureMode.locked);
  }

  @Test
  public void getValueForString_returnsNullForNonexistantValue() {
    assertEquals(
        "Returns null for 'nonexistant'", ExposureMode.getValueForString("nonexistant"), null);
  }

  @Test
  public void toString_returnsCorrectValue() {
    assertEquals("Returns 'auto' for ExposureMode.auto", ExposureMode.auto.toString(), "auto");
    assertEquals(
        "Returns 'locked' for ExposureMode.locked", ExposureMode.locked.toString(), "locked");
  }
}

File: .pub-cache/hosted/pub.dartlang.org/camera_platform_interface-2.2.0/test/types/exposure_mode_test.dart
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:camera_platform_interface/camera_platform_interface.dart';
import 'package:camera_platform_interface/src/types/exposure_mode.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  test('ExposureMode should contain 2 options', () {
    const List<ExposureMode> values = ExposureMode.values;

    expect(values.length, 2);
  });

  test('ExposureMode enum should have items in correct index', () {
    const List<ExposureMode> values = ExposureMode.values;

    expect(values[0], ExposureMode.auto);
    expect(values[1], ExposureMode.locked);
  });

  test('serializeExposureMode() should serialize correctly', () {
    expect(serializeExposureMode(ExposureMode.auto), 'auto');
    expect(serializeExposureMode(ExposureMode.locked), 'locked');
  });

  test('deserializeExposureMode() should deserialize correctly', () {
    expect(deserializeExposureMode('auto'), ExposureMode.auto);
    expect(deserializeExposureMode('locked'), ExposureMode.locked);
  });
}
